src/vehicle.cpp

```diff
@@ -3572,6 +3618,63 @@ void vehicle::power_parts()
         }
     }
 
+    if( has_part( "BOILER", true ) ) {
+        for( size_t b=0 ; b < boilers.size(); ++b ) {
+            printf("\nchecking boiler %zu\n", b);
+            if( !is_boiler_on( b ) ) {
+                printf("boiler is off\n");
+                continue;
```

Comment:
If you already go through all boilers in a loop, you should drop the `has_part` above, which goes through all parts, not just boilers.